pkg/endpoint/restore.go

```diff
@@ -155,6 +112,55 @@ func ReadEPsFromDirNames(ctx context.Context, owner regeneration.Owner, policyGe
 	return possibleEPs
 }
 
+// findEndpointState finds the JSON representation of an endpoint's state in
+// a directory.
+//
+// It prefers reading from the endpoint state JSON file and falls back to
+// reading from the header.
+func findEndpointState(dir string, log *logrus.Entry) ([]byte, error) {
+	state, err := os.ReadFile(filepath.Join(dir, common.EndpointStateFileName))
+	if err == nil {
+		log.Debug("Restoring from JSON file")
```

Comment:
nit: should this be outside the `if` because we are trying to restore from JSON file by default ? or maybe change it to `restored` instead of restoring.